frontend/src/app/testQueue/QueueItem.tsx

```diff
@@ -45,12 +45,12 @@ const EDIT_QUEUE_ITEM = gql`
 interface EditQueueItemParams {
   id: string;
   deviceId?: string;
-  result?: string;
```

Comment:
Defining a lot of global types in the `react-app-env.d.ts` file makes types hard to trace, especially if others follow this pattern for all their types. It took me a while to find where it was defined, I expected the type to be imported directly by the file. Can we create a project policy for when and where types are defined? At the moment we still have a lot of the JS-style `constants.js` pattern which should not be needed in a TS project.

Also noting that this type is more of a hint and not guaranteed to be correct unless we are also runtime-validating the related values that come from the backend. I think that's fine though because at least this prevents the front-end from making a dumb mistake, but it's something to think about if the values delivered by the backend change over time.

Reply:
> Defining a lot of global types in the react-app-env.d.ts file makes types hard to trace, especially if others follow this pattern for all their types. It took me a while to find where it was defined, I expected the type to be imported directly by the file. Can we create a project policy for when and where types are defined? At the moment we still have a lot of the JS-style constants.js pattern which should not be needed in a TS project.

happy to place the types where ever the team wants

> Also noting that this type is more of a hint and not guaranteed to be correct unless we are also runtime-validating the related values that come from the backend. I think that's fine though because at least this prevents the front-end from making a dumb mistake, but it's something to think about if the values delivered by the backend change over time.

I think this can be solved long term by creating a more robust data layer between our view and Apollo where what ever value we can't pull from the schema file we cast before it gets to the view

Reply:
It seems like the types would be somewhere in the vicinity of that new data layer. For now maybe put `TestResult` in this file and import where needed? Once most files are TS I don't find that I have to explicitly mention types that often, but when TS and JS are calling each other it often requires explicit types which is a pain.

Reply:
moved `TestResult` to `QueueItem.tsx`